homeassistant/components/bluetooth/base_scanner.py

```diff
@@ -361,29 +346,39 @@ def _async_on_advertisement(
             device._rssi = rssi  # deprecated, will be removed in newer bleak
         else:
             device = BLEDevice(
-                address=address,
-                name=local_name,
-                details=self._details | details,
-                rssi=rssi,  # deprecated, will be removed in newer bleak
+                address,
+                local_name,
+                self._details | details,
+                rssi,  # deprecated, will be removed in newer bleak
             )
+
+        advertisement_data = AdvertisementData(
+            None if local_name == "" else local_name,
+            manufacturer_data,
+            service_data,
+            service_uuids,
+            NO_RSSI_VALUE if tx_power is None else tx_power,
+            rssi,
+            (),
+        )
         self._discovered_device_advertisement_datas[address] = (
             device,
             advertisement_data,
         )
         self._discovered_device_timestamps[address] = now
         self._new_info_callback(
             BluetoothServiceInfoBleak(
```

Comment:
Honestly, I don't think replacing keywords with positional arguments is an improvement in any Python code...

Reply:
That was a bit too aggressive. Adjusted in 06772b7f79b2d7e2952667afc06eb06b767ef932